custom_components/ha_washdata/suggestion_engine.py

```
@@ -221,15 +291,160 @@ def run_simulation(self, cycle_data: dict[str, Any]) -> dict[str, Any]:
+
+            # False-end energies: low-power segments that resumed
+            in_pause = False
+            pause_energy = 0.0
+            stop_w = 2.0
+            _MAX_PAUSE_GAP_H = 1.0
+            for i in range(1, len(readings)):
+                t0, p0 = readings[i - 1]
+                t1, p1 = readings[i]
+                dt_s = t1 - t0
+                # Guard against non-positive or excessively large time gaps
+                if dt_s <= 0 or dt_s > max_gap_s:
```

Ruff (0.15.10)

[error] Undefined name `max_gap_s`

(F821)